### PR TITLE
feature(component): Rewrite Progress to use themes, resolves #141

### LINKED ISSUE #141: Theme: Progress component
**Is your feature request related to a problem? Please describe.**
Improve the theme support to the component (and subcomponents).

**Describe the solution you'd like**
- All default styles must be moved to the default theme;

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by mouracamila**
### Breaking changes

**Progress** can be customized using  **< Flowbite theme={..} >**

```
  progress: {
    base: string;
    label: string;
    bar: string;
    color: string;
    size: string;
  };
```

### CODE REVIEW

**Comment by mouracamila** [comment]
please @tulup-conner Could you take a look this?  (around the a11y)

**Review by tulup-conner — CHANGES_REQUESTED** [review]

File: src/lib/components/Progress/index.tsx [hunk 1]
```diff
@@ -1,71 +1,51 @@
...
-          <div className="mb-1 flex justify-between font-medium dark:text-white">
+          <div className={theme.label}>
             <span>{label}</span>
             {labelProgress && <span>{progress}%</span>}
           </div>
         )}
-        <div
-          className={classNames('w-full overflow-hidden rounded-full bg-gray-200 dark:bg-gray-700', sizeClasses[size])}
-        >
+        <div className={classNames(theme.base, theme.size[size])}>
```
Comment (rluders) [inline src/lib/components/Progress/index.tsx:46]: You can remove the `classNames` function here.

File: src/lib/components/Progress/index.tsx [hunk 2]
```diff
@@ -1,71 +1,51 @@
...
           <div
-            className={classNames(
-              'flex items-center justify-center rounded-full text-center font-medium leading-none text-blue-100',
-              colorClasses[color],
-              sizeClasses[size],
-              className,
-            )}
+            className={classNames(theme.bar, theme.color[color], theme.size[size])}
```
Comment (rluders) [inline src/lib/components/Progress/index.tsx:48]: You can remove the `classNames` function here.

File: src/lib/components/Progress/index.tsx [hunk 3]
```diff
@@ -1,71 +1,51 @@
 import classNames from 'classnames';
```
Comment (rluders) [inline src/lib/components/Progress/index.tsx:1]: You don't need the classNames import

**Review by rluders — CHANGES_REQUESTED** [review]

**Pull request merged**